From 65f03e01141bec54ea879765eb18f3d0a0e76800 Mon Sep 17 00:00:00 2001
From: Subhadeep Bhandari
 <49233-SubhadeepBhandari@users.noreply.drupalcode.org>
Date: Tue, 4 Oct 2022 19:10:45 -0300
Subject: [PATCH 1/2] Issue #3269083 by S_Bhandari: TypeError: Unsupported
 operand types: string - int in
 Drupal\views_ef_fieldset\Plugin\views\display_extender\ViewsEFFieldset->buildOptionsForm()

---
 src/Plugin/views/display_extender/ViewsEFFieldset.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Plugin/views/display_extender/ViewsEFFieldset.php b/src/Plugin/views/display_extender/ViewsEFFieldset.php
index 015d26e..489a191 100644
--- a/src/Plugin/views/display_extender/ViewsEFFieldset.php
+++ b/src/Plugin/views/display_extender/ViewsEFFieldset.php
@@ -198,7 +198,7 @@ class ViewsEFFieldset extends DefaultDisplayExtender {
 
       $data[] = [
         'id' => $field['id'],
-        'weight' => isset($field_options['weight']) ? $field_options['weight'] : (isset($field->weight) ? $field->weight : $index - count($exposed_fields)),
+        'weight' => $field_options['weight']) ?? $field->weight ?? intval($index) - count($exposed_fields),
         'pid' => empty($field_options['pid']) ? 'root' : $field_options['pid'],
         'label' => $label,
         'title' => isset($field_options['title']) ? $field_options['title'] : $label,
-- 
GitLab


From 4435cb499dd9629535e70650df6b71bce07a7e5b Mon Sep 17 00:00:00 2001
From: osopolar <osopolar@115856.no-reply.drupal.org>
Date: Tue, 4 Oct 2022 22:26:33 +0000
Subject: [PATCH 2/2] #3269083 Fix typo.

---
 src/Plugin/views/display_extender/ViewsEFFieldset.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Plugin/views/display_extender/ViewsEFFieldset.php b/src/Plugin/views/display_extender/ViewsEFFieldset.php
index 489a191..eaf31fd 100644
--- a/src/Plugin/views/display_extender/ViewsEFFieldset.php
+++ b/src/Plugin/views/display_extender/ViewsEFFieldset.php
@@ -198,7 +198,7 @@ class ViewsEFFieldset extends DefaultDisplayExtender {
 
       $data[] = [
         'id' => $field['id'],
-        'weight' => $field_options['weight']) ?? $field->weight ?? intval($index) - count($exposed_fields),
+        'weight' => $field_options['weight'] ?? $field->weight ?? intval($index) - count($exposed_fields),
         'pid' => empty($field_options['pid']) ? 'root' : $field_options['pid'],
         'label' => $label,
         'title' => isset($field_options['title']) ? $field_options['title'] : $label,
-- 
GitLab

